### PR TITLE
Updates for ITKv5

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

**Comment by blowekamp** [comment]
These first changes where the minimal to get things compiling and working.

**Comment by blowekamp** [comment]
@dzenanz 
It would be great to get your input on how to update the [ImageSinc](https://github.com/blowekamp/itkStreamingSinc/blob/master/include/itkImageSinc.h) base class to use the new threading in ITK. This filter base class first streams input regions, then it uses the old threading model to divide the *input* into regions.

Any ideas how to use the new threading on *input* regions?

**Comment by dzenanz** [comment]
> Any ideas how to use the new threading on input regions?

Looking into it.

File: include/itkImageSinc.hxx [hunk 1]
```diff
@@ -173,10 +173,10 @@ ImageSinc<TInputImage>
 {
   ThreadStruct *str;
 
-  const ThreadIdType threadId = ( (MultiThreader::ThreadInfoStruct *)( arg ) )->ThreadID;
-  const ThreadIdType  threadCount = ( (MultiThreader::ThreadInfoStruct *)( arg ) )->NumberOfThreads;
+  const ThreadIdType threadId = ( (MultiThreaderBase::ThreadInfoStruct *)( arg ) )->ThreadID;
```
Comment (dzenanz) [inline include/itkImageSinc.hxx]: ThreadInfoStruct -> WorkUnitInfo

ThreadInfoStruct is deprecated

**Review by dzenanz — COMMENTED** [review]
I was just checking what was being done in this PR.

**Review by dzenanz — APPROVED** [review]

**Pull request merged**